R/tm_variable_browser.R

```diff
@@ -1335,3 +1339,17 @@ remove_outliers_from <- function(var, outlier_definition) {
   iqr <- q1_q3[2] - q1_q3[1]
   var[var >= q1_q3[1] - outlier_definition * iqr & var <= q1_q3[2] + outlier_definition * iqr]
 }
+
+#' Imputes empty strings as `NA`
+#'
+#' @param var (`vector`) a vector of any type and length
+#' @returns (`vector`) a vector with empty strings imputed as `NA`, if provided.
+#' @keywords internal
+impute_blanks_as_na <- function(var) {
+  var <- as.vector(var)
```

Comment:
Dropping `factor`?

Reply:
Yeah, this was the main issue that was raised. If the factor has a level that is "" it will NOT be considered as NA, dropping the factor was the only way to consider them as NA.
I'm still keeping the PR as a draft because the request is unclear if we need to be doing this.

Reply:
Will this not influence the plotting, like change the order of categories on the axis or in the color key?

Reply:
We explicitly change the NAs to `<Missing>` before we plot them. Previously the label was `NA` with a different color. Now, we're just displaying it as `<Missing>` because it is in line with the `tern::df_explicit_na()`

```r
r$> df_explicit_na(data.frame(col = c("A", "B", NA, "C")))
        col
1         A
2         B
3 <Missing>
4         C
```

![image](https://github.com/insightsengineering/teal.modules.general/assets/49812166/7e35ccb8-b147-4822-bb3f-24597126d872)

Again, so much of a feature request here. Nothing is clear as of now. So, it's still a draft and we can talk about what we want.